widget/editor.go

```diff
@@ -376,8 +545,8 @@ func (e *Editor) Layout(gtx layout.Context, sh text.Shaper, font text.Font, size
 	if viewSize := gtx.Constraints.Constrain(e.dims.Size); viewSize != e.viewSize {
 		e.viewSize = viewSize
 		e.invalidate()
+		e.makeValid()
 	}
-	e.makeValid()
```

Comment:
Is this a performance optimization? If so, please split it out into a separate change; makeValid calls have historically been very tricky to get right, and a separate change for this move will make future git bisections easier.